src/test/java/seedu/address/logic/commands/ListFavouritesCommandTest.java

```diff
@@ -0,0 +1,62 @@
+package seedu.address.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static seedu.address.commons.core.Messages.MESSAGE_PERSONS_LISTED_OVERVIEW;
+import static seedu.address.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.address.testutil.TypicalPersons.BENSON;
+import static seedu.address.testutil.TypicalPersons.FIONA;
+import static seedu.address.testutil.TypicalPersons.getTypicalAddressBook;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import seedu.address.model.Model;
+import seedu.address.model.ModelManager;
+import seedu.address.model.UserPrefs;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.PersonIsFavouriteContactPredicate;
+import seedu.address.testutil.PersonBuilder;
+
+class ListFavouritesCommandTest {
+
+    private Model model;
+    private Model expectedModel;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalAddressBook(), new UserPrefs());
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_multiplePersonsFound() {
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 2);
+        PersonIsFavouriteContactPredicate predicate = new PersonIsFavouriteContactPredicate();
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(new ListFavouritesCommand(), model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(BENSON, FIONA), model.getFilteredPersonList());
+    }
+
+    @Test
+    public void execute_noPersonFound() {
+        // unfavourite all favourite persons
+        List<Person> typicalPersons = model.getFilteredPersonList();
+        for (Person originalPerson : typicalPersons) {
+            if (originalPerson.getFavouriteStatus().isFavourite()) {
+                Person unfavouritedPerson = new PersonBuilder(originalPerson).withFavourite("false").build();
+                model.setPerson(originalPerson, unfavouritedPerson);
+            }
+        }
+        expectedModel = new ModelManager(model.getAddressBook(), new UserPrefs());
+
+        String expectedMessage = String.format(MESSAGE_PERSONS_LISTED_OVERVIEW, 0);
+        PersonIsFavouriteContactPredicate predicate = new PersonIsFavouriteContactPredicate();
+        expectedModel.updateFilteredPersonList(predicate);
+        assertCommandSuccess(new ListFavouritesCommand(), model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredPersonList());
+    }
+}
```

Comment:
Since we have tests for multiple persons and no person at all, would it be more complete to test if a single person is found as well?

Reply:
I see where you're coming from! Here's what I'm testing for:
1. Does the correct number get reflected on the message based on the number of favourited people? (Multiple, Zero)
2. Does the resulting list look like a list we would expect? For example, empty list if no favourited, and a list with exactly the 2 people we favourited? (Multiple, Zero)

I think that adding a single person test doesn't might be redundant since it doesn't add-on to the existing test cases, let me know if I missed to account for something in my cases 😄